.github/workflows/dotnet.yml

```diff
@@ -20,6 +19,15 @@ jobs:
     - name: Restore dependencies
       run: dotnet restore
     - name: Build
-      run: dotnet build --no-restore
+      run: dotnet build --no-restore -c Release -o ./output
     - name: Test
       run: dotnet test --no-build --verbosity normal
+
+  publish:
+    if: github.ref == 'refs/heads/main'
```

Comment:
💭 Should probably check the organisation and (potentially) repository name as well. This workflow shouldn't run in a fork.

Reply:
Also, have you considered running this workflow job for tag events instead? That way we can be sure that a version has been tagged in version control before it is published to NuGet.

Reply:
> 💭 Should probably check the organisation and (potentially) repository name as well. This workflow shouldn't run in a fork.

This would fail if run on a fork because there is no nuget key secret specified in the fork. If they did specify the secret in their repository then the push would fail because the owner of the secret is not in duffel org.

Reply:
> Also, have you considered running this workflow job for tag events instead? That way we can be sure that a version has been tagged in version control before it is published to NuGet.

Yeah, I was thinking about this as otherwise the build will fail if we merge to main without bumping version. Your suggestion would be a better flow. I'll look into making it publish when a new tag is pushed instead of any push.

Reply:
Ah, thanks @sgerrand i'll have a look at that haden't seen it